orangecontrib/educational/widgets/owkmeans.py

```diff
@@ -0,0 +1,532 @@
+
+    def button_text_change(self):
+        """
+        Function changes text on ste button and chanbe the button text
+        """
+        self.stepButton.setText(self.button_labels["step2"]
+                                if self.k_means.step_completed
+                                else self.button_labels["step1"])
+        if self.k_means.stepNo <= 0:
+            self.stepBackButton.setDisabled(True)
+        elif not self.autoPlay:
+            self.stepBackButton.setDisabled(False)
+
+    def auto_play(self):
+        """
+        Function called when autoplay button pressed
+        """
+        self.autoPlay = not self.autoPlay
+        self.autoPlayButton.setText(self.button_labels["autoplay_stop"]
+                                    if self.autoPlay
+                                    else self.button_labels["autoplay_run"])
+        if self.autoPlay:
+            self.optionsBox.setDisabled(True)
+            self.centroidsBox.setDisabled(True)
+            self.stepButton.setDisabled(True)
+            self.stepBackButton.setDisabled(True)
+            self.autoPlayThread = Autoplay(self)
+            self.connect(self.autoPlayThread, SIGNAL("step()"), self.step)
+            self.connect(self.autoPlayThread, SIGNAL("stop_auto_play()"), self.stop_auto_play)
+            self.autoPlayThread.start()
+        else:
+            self.stop_auto_play()
+
+    def stop_auto_play(self):
+        """
+        Called when stop autoplay button pressed or in the end of autoplay
+        """
+        self.optionsBox.setDisabled(False)
+        self.stepButton.setDisabled(False)
+        self.centroidsBox.setDisabled(False)
+        self.stepBackButton.setDisabled(False)
+        self.autoPlay = False
+        self.autoPlayButton.setText(self.button_labelsp["autoplay_stop"]
+                                    if self.autoPlay
+                                    else self.button_labels["autoplay_run"])
+
+    def replot(self):
+        """
+        Function refreshes the chart
+        """
+        if self.data is None or not self.attr_x or not self.attr_y:
+            return
+
+        if self.k_means.centroids_moved:
+            # when centroids moved during step
+            self.scatter.update_series(0, self.k_means.centroids)
+
+            if self.lines_to_centroids:
+                for i, c in enumerate(self.k_means.centroids):
+                    self.scatter.update_series(1 + i, list(chain.from_iterable(
+                        ([p[0], p[1]], [c[0], c[1]])
+                        for p in self.k_means.centroids_belonging_points[i])))
+        else:
+            self.complete_replot()
+
+    def complete_replot(self):
+        """
+        This function performs complete replot of the graph without animation
+        """
+        attr_x, attr_y = self.data.domain[self.attr_x], self.data.domain[self.attr_y]
+
+        # plot centroids
+        options = dict(series=[])
+        options['series'].append(dict(data=[{'x': p[0],
```

Comment:
If you make a new line before data=, you can reduce the empty space on the left.